store/localcachelayer/channel_layer.go

```diff
@@ -73,6 +87,56 @@ func (s LocalCacheChannelStore) InvalidateMemberCount(channelId string) {
 	}
 }
 
+// removeByPrefixFromChannelMembersForUserCache removes all keys from
+// channelMembersForUserCache with the prefix of the given userId.
+func (s LocalCacheChannelStore) removeByPrefixFromChannelMembersForUserCache(userId string) {
+	keys := s.rootStore.channelMembersForUserCache.Keys()
+	for _, key := range keys {
+		keyString := key.(string)
+		if strings.HasPrefix(keyString, userId) {
+			s.rootStore.channelMembersForUserCache.Remove(keyString)
+		}
+	}
+}
+
+// InvalidateMembersForUser removes all keys from channelMembersForUserCache
+// with the prefix of the given userID.
+// We cannot simply call doInvalidateCacheCluster because we need to remove
+// keys with a prefix rather than remove a single key.
+func (s LocalCacheChannelStore) InvalidateMembersForUser(userId string) {
+	s.removeByPrefixFromChannelMembersForUserCache(userId)
+	if s.rootStore.cluster != nil {
+		msg := &model.ClusterMessage{
+			Event:    model.CLUSTER_EVENT_INVALIDATE_CACHE_FOR_CHANNEL_MEMBERS_FOR_USER,
+			SendType: model.CLUSTER_SEND_BEST_EFFORT,
+			Data:     userId,
+		}
+		s.rootStore.cluster.SendClusterMessage(msg)
+	}
+
+	if s.rootStore.metrics != nil {
+		s.rootStore.metrics.IncrementMemCacheInvalidationCounter("Channel Members For User - Remove by UserId")
+	}
+}
+
+// InvalidateMembersForAllUsers purges the entire channelMembersForUserCache cache.
+func (s LocalCacheChannelStore) InvalidateMembersForAllUsers() {
```

Comment:
Because this is not used outside the cache layer, this doesn't need to be exposed to the outside, can be a private method, and not included in the `Store` interface (so you can remove the empty implementation in the `SqlStore`)

Reply:
Yep :+1: